libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -384,6 +382,10 @@ def mmr_traversal_search(
 
         Args:
             query: The query string to search for.
+            neighborhood: Optional list of documents to use as the initial neighborhood
```

Comment:
Note: Currently, providing a neighborhood bypasses the initial fetch. This means that no new "root" nodes will be fetched. This may be a good thing, but I could also be convinced that we should take the union of the neighborhood-adjacencies and the `fetch-k`, and allow `fetch_k = 0` to indicate "only use the neighborhood)... thoughts?

Reply:
Also, don't love the name...